src/shared/models/game/Effect.es6

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+//
+// Effect
+// Created by dpekar on 10/12/15.
+//
+
+const _ = require('underscore')
+const Logger = require('../../log/Logger')
+
+const BaseModel = require('../BaseModel')
+// const StatsConfig = require('../../data/Stats-config')
+const TileConfig = require('../../data/Tile-config')
+
+class Effect extends BaseModel {
+    constructor(effectInfo) {
+        super()
+        this.setFromObject(effectInfo)        
+    }
+
+    getStat() { return this.get('stat') }
+    getValue() { return this.get('value') }
+    getCondition() { return this.get('condition') }
+
+    executeIf(player, newPlacement, existingPlacement) {
+        let condition = this.getCondition()
+        let tile = newPlacement.getTile()
+        if (tile.meetsCondition(condition)) {
+            Logger.info(`${tile.getName()} because of ${existingPlacement.getTile().getName()} produces ${this.getValue()} ${this.getStat()}`)
+            if(existingPlacement.getInvestedIn() === true) {
+                player.incrementStat(this.getStat(), this.getValue())
+            }  // Execute the effect twice if the person invested in the tile already
+            player.incrementStat(this.getStat(), this.getValue())
+        }
+    }
+
+    executeOn(player, placement, gameState) {
```

Comment:
I need to create subclasses for effects to get rid of this nasty switch statement.